### PR TITLE
Added demo for flyout component

### PR DESCRIPTION
**Pull request by Blackbaud-AlexKingman**
We'll need to fix this to get the jsDocs to show back up: https://github.com/blackbaud/skyux-docs-tools/issues/111 It shouldn't block this PR though.

### CODE REVIEW

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-flyout?_sp.spa=pr-flyout-demo](https://developer.blackbaud.com/skyux-flyout?_sp.spa=pr-flyout-demo)

File: src/app/docs/flyout/flyout-docs-flyout.component.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      class="sky-padding-even-large"
+    >
+      <h2>
+        Sample flyout
+      </h2>
+      <p>
+        Flyouts can display large quantities of supplementary information related to a task, including:
+      </p>
+        <ul>
+          <li><a href="https://developer.blackbaud.com/skyux/components/list">lists</a></li>
```
Comment (blackbaud-johnly) [inline src/app/docs/flyout/flyout-docs-flyout.component.ts:18]:
```suggestion
          <li>lists</li>
```
Seems like we should get rid of this link and avoid steering users toward that module given that we want to de-emphasize the module in preference for data entry grid or other options.

File: src/app/docs/flyout/flyout-docs-flyout.component.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+          <li>record views</li>
```
Comment (blackbaud-johnly) [inline src/app/docs/flyout/flyout-docs-flyout.component.ts:19]:
```suggestion
          <li>records</li>
```

File: src/app/docs/flyout/flyout-docs-flyout.component.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+          <li>analytics</li>
+          <li>other information</li>
```
Comment (blackbaud-johnly) [inline src/app/docs/flyout/flyout-docs-flyout.component.ts:21]:
```suggestion
```

File: src/app/docs/flyout/flyout-docs-flyout.component.ts (hunk 1 above)
Comment (Blackbaud-AlexKingman) [reply src/app/docs/flyout/flyout-docs-flyout.component.ts:18]: Yep, good idea.

File: src/app/public/plugin-resources/code-examples/flyout/basic/flyout-demo-flyout.component.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+      class="sky-padding-even-large"
+    >
+      <h2>
+        Sample flyout
+      </h2>
+      <p>
+        Flyouts can display large quantities of supplementary information related to a task, including:
+      </p>
+        <ul>
+          <li><a href="https://developer.blackbaud.com/skyux/components/list">lists</a></li>
```
Comment (blackbaud-johnly) [inline src/app/public/plugin-resources/code-examples/flyout/basic/flyout-demo-flyout.component.ts:18]:
```suggestion
          <li>lists</li>
```

File: src/app/public/plugin-resources/code-examples/flyout/basic/flyout-demo-flyout.component.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+          <li>record views</li>
```
Comment (blackbaud-johnly) [inline src/app/public/plugin-resources/code-examples/flyout/basic/flyout-demo-flyout.component.ts:19]:
```suggestion
          <li>records</li>
```

File: src/app/public/plugin-resources/code-examples/flyout/basic/flyout-demo-flyout.component.ts [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+          <li>analytics</li>
+          <li>other information</li>
```
Comment (blackbaud-johnly) [inline src/app/public/plugin-resources/code-examples/flyout/basic/flyout-demo-flyout.component.ts:21]:
```suggestion
```

File: src/app/public/plugin-resources/code-examples/flyout/custom-headers/flyout-demo-flyout.component.ts [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+      class="sky-padding-even-large"
+    >
+      <h2>
+        Sample flyout
+      </h2>
+      <p>
+        Flyouts can display large quantities of supplementary information related to a task, including:
+      </p>
+        <ul>
+          <li><a href="https://developer.blackbaud.com/skyux/components/list">lists</a></li>
```
Comment (blackbaud-johnly) [inline src/app/public/plugin-resources/code-examples/flyout/custom-headers/flyout-demo-flyout.component.ts:18]:
```suggestion
          <li>lists</li>
```

File: src/app/public/plugin-resources/code-examples/flyout/custom-headers/flyout-demo-flyout.component.ts [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+          <li>record views</li>
```
Comment (blackbaud-johnly) [inline src/app/public/plugin-resources/code-examples/flyout/custom-headers/flyout-demo-flyout.component.ts:19]:
```suggestion
          <li>records</li>
```

File: src/app/public/plugin-resources/code-examples/flyout/custom-headers/flyout-demo-flyout.component.ts [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+          <li>analytics</li>
+          <li>other information</li>
```
Comment (blackbaud-johnly) [inline src/app/public/plugin-resources/code-examples/flyout/custom-headers/flyout-demo-flyout.component.ts:21]:
```suggestion
```

File: src/app/public/plugin-resources/code-examples/flyout/custom-headers/flyout-demo.component.ts [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+
+    this.flyout.closed.subscribe(() => {
+      this.flyout = undefined;
+    });
+  }
+
+  public openFlyoutWithUrlPermalink(): void {
+    const flyoutConfig: SkyFlyoutConfig = {
+      permalink: {
+        label: `Visit blackbaud.com`,
```
Comment (blackbaud-johnly) [inline src/app/public/plugin-resources/code-examples/flyout/custom-headers/flyout-demo.component.ts:71]: In the demo, when I click this button, it opens blackbaud.com. But in Stackblitz, it just closes the flyout. Is this the expected behavior?

**Review by blackbaud-johnly — COMMENTED** [review]
A few comments and suggestions... I checked the code samples and tested them in Stackblitz, but it might be good to have a developer check too. ... Also, I noticed that the PR preview is not displaying the development properties, but when I just go to https://developer.blackbaud.com/skyux-flyout/docs/flyout, those properties are in place. Is your branch out of sync with `master`?

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-flyout?_sp.spa=pr-flyout-demo](https://developer.blackbaud.com/skyux-flyout?_sp.spa=pr-flyout-demo)

File: src/app/public/plugin-resources/code-examples/flyout/custom-headers/flyout-demo.component.ts (hunk 10 above)
Comment (Blackbaud-AlexKingman) [reply src/app/public/plugin-resources/code-examples/flyout/custom-headers/flyout-demo.component.ts:71]: Yeah, stackblitz prevents the link from firing for some reason. It also doesn't work for router params.

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-flyout?_sp.spa=pr-flyout-demo](https://developer.blackbaud.com/skyux-flyout?_sp.spa=pr-flyout-demo)

**Review by blackbaud-johnly — APPROVED** [review]

**Review by Blackbaud-DeniseCisneros — APPROVED** [review]

**Review by Blackbaud-SteveBrush — APPROVED** [review]

**Comment by Blackbaud-AlexKingman** [comment]
> Also, I noticed that the PR preview is not displaying the development properties, but when I just go to https://developer.blackbaud.com/skyux-flyout/docs/flyout, those properties are in place. Is your branch out of sync with `master`?

@blackbaud-johnly That's a known issue: https://github.com/blackbaud/skyux-docs-tools/issues/111

**Pull request merged**